File: ArtificialIntelligence/assignment4/reversi/random.lua
function do_move(board, color)
    local moves = board:valid_moves( color )
    local len = #moves

    if len == 0 then
        return nil
    end

    local idx = math.random(len)
    local item = moves[ idx ]

    return item[ 1 ], item[ 2 ]
end

File: ArtificialIntelligence/assignment4/reversi/src/player/random.rs
use player::Player;
use board::{Board, Color};

use rand::{self, Rng};

pub struct RandomPlayer;

impl Player for RandomPlayer {
    fn do_move( &mut self, board: & Board, color: Color ) -> Option<(usize, usize)> {
        let moves = board.valid_moves( color );
        rand::thread_rng().choose( &moves ).cloned()
    }
}
